network/src/connectivity_manager/builder.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    connectivity_manager::{ConnectivityManager, ConnectivityRequest},
+    counters,
+    peer_manager::{conn_notifs_channel, ConnectionRequestSender},
+};
+use channel::{self};
+use futures::stream::StreamExt;
+use futures_util::stream::Fuse;
+use libra_config::network_id::NetworkContext;
+use libra_crypto::x25519;
+use libra_network_address::NetworkAddress;
+use libra_types::PeerId;
+use std::{
+    collections::HashMap,
+    sync::{Arc, RwLock},
+    time::Duration,
+};
+use tokio::{
+    runtime::Handle,
+    time::{interval, Interval},
+};
+use tokio_retry::strategy::ExponentialBackoff;
+
+pub type ConnectivityManagerService = ConnectivityManager<Fuse<Interval>, ExponentialBackoff>;
+
+/// The configuration fields for ConnectivityManager
+struct ConnectivityManagerBuilderConfig {
+    network_context: Arc<NetworkContext>,
+    eligible: Arc<RwLock<HashMap<PeerId, x25519::PublicKey>>>,
+    seed_peers: HashMap<PeerId, Vec<NetworkAddress>>,
+    connectivity_check_interval_ms: u64,
+    backoff_base: u64,
+    max_connection_delay_ms: u64,
+    connection_reqs_tx: ConnectionRequestSender,
+    connection_notifs_rx: conn_notifs_channel::Receiver,
+    requests_rx: channel::Receiver<ConnectivityRequest>,
+    connection_limit: Option<usize>,
+}
+
+#[derive(Debug, PartialEq, PartialOrd)]
+enum State {
+    CREATED,
+    BUILT,
+    STARTED,
+}
```

Comment:
It may help to share enums, and create a trait that each of these implement at least for start/build.

Reply:
I thought about that.  and it is probably right, but there is a very complicated bikeshed to be had about *where* that trait lives.

let's not bite that problem off quite yet.

Reply:
Yeah, right now it's fine.  Let's just think about it in the future.